### PR TITLE
feat: add TypeScript type annotation tests with node 22 snapshots

### PR DESCRIPTION
**Pull request by mertcanaltin**
Greetings, I added a test and tried to get more coverage

- Add test for stripping type annotations from functions
- Add test for stripping type annotations from classes
- Add test for stripping type annotations from interfaces
- Add test for stripping type annotations from type aliases
- Add test for stripping type annotations from generics
- Add test for stripping type annotations from arrow functions
- Add test for stripping type annotations from type assertions

Additionally, I have integrated node 22 snapshot testing as suggested. The snapshot tests have been normalized to handle line endings and formatting issues

### CODE REVIEW

**Comment by marco-ippolito** [comment]
Maybe we can use node 22 snapshots testing?

**Comment by mertcanaltin** [comment]
> Maybe we can use node 22 snapshots testing?

do you think I should integrate it here?

**Comment by marco-ippolito** [comment]
> > Maybe we can use node 22 snapshots testing?
> 
> do you think I should integrate it here?

Id give it a try, and we can use node 22 in the CI

**Comment by mertcanaltin** [comment]
Hi, I converted it to a snapshot

**Comment by marco-ippolito** [comment]
missing EOF and some formatting issues

File: test/snapshot.js [hunk 1]
```diff
@@ -1,33 +1,38 @@
-const fs = require("node:fs");
-const path = require("node:path");
+const fs = require("fs");
```
Comment (marco-ippolito) [inline test/snapshot.js]: please use `node:` namespace for imports

Comment (mertcanaltin) [reply test/snapshot.js]: ı used, thanks

File: package.json [hunk 2]
```diff
@@ -23,7 +23,7 @@
 		"postpack": "npm run clean",
 		"build": "rspack build",
 		"typecheck": "tsc --noEmit",
-		"test": "node --test ./test"
+		"test": "node --test --experimental-test-snapshots --test-update-snapshots --require ./test/snapshot-config.js ./test/index.test.js"
```
Comment (marco-ippolito) [inline package.json:26]: the creation of snapshots should be run in a different command like
`test:regenerate`

**Comment by mertcanaltin** [comment]
I updated the test structure, it seemed that there was no need for the main index test anymore, so I updated the test commands in the package.json, but if any tests are added, we can update here

**Comment by AugustinMauroy** [comment]
I think we can add test for this code:
```ts
type User = {
  name: string;
  age: number;
};

function isAdult(user: User): boolean {
  return user.age >= 18;
}

const justine = {
  name: 'Justine',
  age: 23,
} satisfies User;

const isJustineAnAdult = isAdult(justine);
```

it will be added to the learn section as an example it would be interesting to test it

File: test/setup.mjs [hunk 3]
```diff
@@ -0,0 +1,4 @@
+import { register } from "node:module";
```
Comment (AugustinMauroy) [inline test/setup.mjs:1]: this file isn't use we can remove it ?

File: test/snapshots/index.test.js [hunk 4]
```diff
@@ -0,0 +1,80 @@
+const { test } = require("node:test");
```
Comment (marco-ippolito) [inline test/snapshots/index.test.js:1]: shouldnt this test be in the root ? also why remove white space? the whole point is to check for whitespaces

File: test/setup.mjs (hunk 3 above)
Comment (mertcanaltin) [reply test/setup.mjs]: Of course, thank you

File: test/snapshots/index.test.js (hunk 4 above)
Comment (mertcanaltin) [reply test/snapshots/index.test.js:1]: I thought we were gonna keep the snapshots tests separate

Comment (marco-ippolito) [reply test/snapshots/index.test.js:1]: The test should stay in the test folder, snapshot output in the test/snapshot folder

Comment (mertcanaltin) [reply test/snapshots/index.test.js]: I tried to fix it

**Comment by mertcanaltin** [comment]
> I think we can add test for this code:
> 
> ```ts
> type User = {
>   name: string;
>   age: number;
> };
> 
> function isAdult(user: User): boolean {
>   return user.age >= 18;
> }
> 
> const justine = {
>   name: 'Justine',
>   age: 23,
> } satisfies User;
> 
> const isJustineAnAdult = isAdult(justine);
> ```
> 
> it will be added to the learn section as an example it would be interesting to test it

I have added a test on this, thanks for the suggestion

File: package.json [hunk 5]
```diff
@@ -23,7 +23,8 @@
 		"postpack": "npm run clean",
 		"build": "rspack build",
 		"typecheck": "tsc --noEmit",
-		"test": "node --test ./test"
+		"test": "node --test --experimental-test-snapshots ./test/index.test.js",
```
Comment (marco-ippolito) [inline package.json]: We want to execute the whole folder rathee then a single file

File: package.json [hunk 6]
```diff
@@ -23,7 +23,8 @@
...
+		"test:regenerate": "node --experimental-test-snapshots --test-update-snapshots ./test/index.test.js"
```
Comment (marco-ippolito) [inline package.json:27]: Same here

Comment (mertcanaltin) [reply package.json:27]:
```js
"test": "node --test ./test"
```

ı used this commant, but get a this error in tests

```js
ℹ tests 1
ℹ suites 0
ℹ pass 0
ℹ fail 1
ℹ cancelled 0
ℹ skipped 0
ℹ todo 0
ℹ duration_ms 86.965917

✖ failing tests:

test at test:1:1
✖ test (79.325333ms)
  'test failed'
➜  amaro git:(dev-1) ✗ 
```

Comment (marco-ippolito) [reply package.json:27]: Thats odd try to remove the path completely

Comment (mertcanaltin) [reply package.json:27]:
```js
"test": "node --test --experimental-test-snapshots",
"test:regenerate": "node --test --experimental-test-snapshots --test-update-snapshots ./test"
```

unfortunately I failed I updated this way

**Comment by marco-ippolito** [comment]
this should work
```
		"test": "node --test --experimental-test-snapshots ./test/*.test.js",
		"test:regenerate": "node --test --experimental-test-snapshots --test-update-snapshots ./test/*.test.js"
```

**Comment by marco-ippolito** [comment]
Ci needs to be run with node 22 and also the engine field with node => 22 in the package.json

File: package.json (hunk 6 above)
Comment (mertcanaltin) [reply package.json]: this problem was solved in your last suggestion thank you

**Comment by mertcanaltin** [comment]
yes it worked as you suggested I tried at 22 🚀

**Comment by marco-ippolito** [comment]
please update the ci.yml to use node 22 and add the engines in the package json

File: .github/workflows/ci.yml [hunk 7]
```diff
@@ -10,19 +10,8 @@ env:
   YARN_ENABLE_GLOBAL_CACHE: false
 
 jobs:
-  commit-lint:
```
Comment (marco-ippolito) [inline .github/workflows/ci.yml:13]: Why remove commit lint, you need to rebase

Comment (mertcanaltin) [reply .github/workflows/ci.yml:13]: I forgot to update the main branch, so sorry, this linti crushed it 😢

Comment (mertcanaltin) [reply .github/workflows/ci.yml]: ı solved

File: test/setup.mjs [hunk 8]
```diff
@@ -0,0 +1,3 @@
+register("some-typescript-loader");
```
Comment (marco-ippolito) [inline test/setup.mjs:1]: Whats the point of this?

Comment (mertcanaltin) [reply test/setup.mjs:1]: I thought it was necessary because it was in the document, but we don't need it, I think I take it back

**Review by marco-ippolito — APPROVED** [review]

**Pull request merged**